### ISSUE TITLE
Unable to Add Ticket Items for Job/Ticket/Pricebook/Pricebook items that user has permissions to

### ISSUE THREAD

**Issue by githubbob42** [githubbob42/mingle2github2#5115]
**[Mingle Card: 5469](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/5469)**
Steps to Reproduce
------------------

```
| | |
|-|-|
|**Version #**|4.0.1.3911|
|**Hardware**|PC Laptop  |
|**OS**|Windows 10|
|**Browser**|Chrome|
|**Username**|lfwtestuser@lfw.com.carbersb|
|**Password**|Fieldfx!1|
|ORG ID|00D1g0000004Y9G|
|User ID|0051g000000NTMT|
|RayGun Error ID| |
  
 

```

1.  Create new test user with mobile version Sync v4
2.  Add Crew Planning for test user to a Job with status ‘In Progress’
3.  Open mobile with test user
4.  Sync
5.  In back office, while logged in as sys admin, add a new ticket to the same job you’ve selected to test against
6.  navigate back to mobile logged in as test user and perform an incremental sync
7.  open the test ticket that you just created
8.  try to add a new ticket item
9.  go back to back office as sys admin and add a new price book item
10.  then go to mobile logged in a test user and Sync

Expected Result
---------------

_You should be able to see the blue + button to add new ticket items in step 7 from above and see all active ticket items for the pricebook on that Job._

Actual Result
-------------

_You don’t see the blue + button. However, if you added any ticket items in back office, you would be able to see them after incremental sync. Also, after step 10, you see the newly created pricebook item, but none of the other pricebook items that the test user has permissions to see. _

_Analysis_
----------

The issue is adding new price book items when a new pricebook gets added to the sync.

There are actually a few bugs in the “price-book-items.js” module but they are all nested so fixing any one doesn’t help.

First, it was pulling the PBI Ids and not the PB Ids in the “getExistingPriceBooks” method.  Next it was setting them to an instance variable and not returning the results but the calling code was expecting the results returned. Then there was fetchJson() and bulkPut() not being referenced properly.

Related Cards
-------------

#5258, #5233

```
| | |
|-|-|
 
  
   
   

   |
  
 

```

Test Plan
---------

It’s probably best (ie. easiest) to test this with only 1 or 2 jobs/pricebooks (I used lfwtestuser2@lfw.com.carbersb/Fieldfx!1 to test, on job **TEST Job - Item Code** in their sandbox env)

1.  Sync mobile
2.  In back-office, add user to a crew on a job with a PB that you do not currently have
3.  sync mobile again
4.  go to a ticket under that new job assignment and go to Item grid
    1.  Add “+” button should there

```
| | |
|-|-|
 
  
   
   

1.

   |
   
   

Ensure the card has enough information from the programmer before you start the verification

   

If not request more information

   |
   
   

   |
  
   
   

2.

   |
   
   

Ensure you’re able to reproduce the defect prior to verifying it

   |
   
   

   |
  
   
   

3.

   |
   
   

Ensure to verify if the PR is still valid by going to Github.

   |
   
   

   |
  
   
   

3.

   |
   
   

Create a test plan and write/update test case for the card is there is no test case in Tarantula.

   |
   
   

   |
  
   
   

4

   |
   
   

Test the card on all required devices and versions. If it’s a mobile card, always test the offline functionality around that defect. Attach screenshots to the card as necessary displaying the fix

   |
   
   

   |
  
   
   

5.

   |
   
   

Add the following test result documentations:

   

    *   Test Status:
    *   PR Build:
    *   Username/Password
    *   Test case name:
    *   Environment and devices tested on:
    *   Test Note.
   

   |
   
   

   |
  
   
   

6.

   |
   
   

Push the card to “Testing Complete”

   |
   
   

   |
  
 

```

QA Test Analysis:

*   Test Status: Passed
*   PR Build: pr2627
*   Username/Password: The credentials displayed on the card
*   Test case name: None
*   Environment and devices tested on: sandbox.fieldfx.com
*   Test Note.: I was able to reproduce this defect by adding a new crew member to a job after a full sync and then performed an incremental sync. the Job was sync down to mobile for the field user but couldn’t add any pricebook items to the job’s ticket.  
    After applying pr2627, i repeated the same scenario and the field user was able to successfully add pricebook items. 

  
!Card\_5469.png!{height: 571px; width: 1200px;}

**Comment by githubbob42** [comment]
**Pull Request #2627**

**Comment by githubbob42** [comment]
Patty : 10/25/2018 10:01 CDT
Documented as bug fix in 1/18/2018 Mobile release notes.
